tests/main.rs

```diff
@@ -22,24 +22,182 @@ trait LendingIterator {
     ;
 }
 
-type Item<'lt, I> = Gat!(<I as LendingIterator>::Item<'lt>);
-
-struct Infinite;
-
 #[gat]
-impl LendingIterator for Infinite {
-    type Item<'next>
+trait MultiLendingIterator {
+    type Item1<'next>
+    where
+        Self : 'next,
+    ;
+    type Item2<'next>
     where
         Self : 'next,
-    =
-        &'next mut Self
     ;
 
     fn next (
         self: &'_ mut Self,
-    ) -> Option<&'_ mut Self>
-    {
-        Some(self)
+    ) -> Option<(Self::Item1<'_>, Self::Item2<'_>)>
+    ;
+}
```

Comment:
Adding a test with multiple GATs is a nice touch, I thank you for it 🙏

Reply:
:100: quality first

also could tell from what you wrote in the issue that you care about users both now and long term :star2:, always appreciate that